src/core/qgslegendrenderer.cpp

```diff
@@ -610,25 +610,13 @@ QgsLegendRenderer::Nucleon QgsLegendRenderer::drawSymbolItemInternal( QgsLayerTr
 {
   QgsLayerTreeModelLegendNode::ItemContext ctx;
   ctx.context = context;
-
-  // add a layer expression context scope
-  QgsExpressionContextScope *layerScope = nullptr;
-  if ( context && symbolItem->layerNode()->layer() )
-  {
-    layerScope = QgsExpressionContextUtils::layerScope( symbolItem->layerNode()->layer() );
-    context->expressionContext().appendScope( layerScope );
-  }
-
```

Comment:
Symbol level evaluation should now be handled through the rest of the added code to prevent duplicate, this section is removed.